cmake/gsl_functions.cmake

```diff
@@ -4,32 +4,19 @@
 # for multiple versions of cmake.
 #
 # Any functions/macros should have a gsl_* prefix to avoid problems
-if (CMAKE_VERSION VERSION_GREATER 3.10 OR CMAKE_VERSION VERSION_EQUAL 3.10)
-    include_guard()
-else()
-    if (DEFINED guideline_support_library_include_guard)
-        return()
-    endif()
-    set(guideline_support_library_include_guard ON)
-endif()
```

Comment:
Since this file is only ever included once. The include_guard() really serves no purpose.